### DISCUSSION TITLE
Request for Static Binary Installation Package for Podman

### DISCUSSION THREAD

**Discussion by ghost** [containers/podman#23743] [General]
### Feature request description

I would like to request the availability of a static binary installation package for Podman. This would be beneficial for the following reasons:

Internal Network Environment: In our environment, we are operating within an internal network, where the dnf installation method can be cumbersome due to numerous RPM dependencies.

Security Concerns with dnf Repositories: The RPM packages available in the dnf repositories tend to be outdated, posing potential security risks. For example, Rocky Linux currently provides version 4.9.4.

Providing a static binary package similar to Docker's (e.g., [Docker Static Binaries](https://download.docker.com/linux/static/stable/x86_64/)) would greatly simplify the installation process and enhance security.

Thank you for considering this request.

### Have you considered any alternatives?

we have considered alternatives, including using Docker. However, we prefer to use Podman due to its advantages in being daemonless and its compatibility with Kubernetes. A static binary package would allow us to utilize Podman in environments where Docker is not an option, without the complications of managing dependencies through dnf.

**Comment by afbjorklund** [comment]
The project decided **not** do to static binaries (except podman-remote), but to leave that up to Linux distributions...
* https://github.com/containers/podman/issues/13298
* https://podman.io/docs/installation#linux-distributions

The CRI-O project still produces a static Nix build, but the Podman project does not feature a static Nix build anymore.
* https://github.com/containers/podman/issues/8788
* https://github.com/containers/podman/issues/5774

**Reply by afbjorklund** [reply]
Probably the Nix package from cri-o can be reused also for podman?

https://github.com/cri-o/cri-o/tree/main/nix/

Something like:

```diff
--- cri-o/nix/derivation.nix	2024-08-19 12:29:49.228843208 +0000
+++ podman/nix/derivation.nix	2024-08-26 12:21:18.243906135 +0000
@@ -2,7 +2,7 @@
 , pkgs
 }:
 with pkgs; buildGo122Module {
-  name = "cri-o";
+  name = "podman";
   # Use Pure to avoid exuding the .git directory
   src = nix-gitignore.gitignoreSourcePure [ ../.gitignore ] ./..;
   vendorHash = null;
@@ -33,7 +33,7 @@
     export CFLAGS='-static -pthread'
     export LDFLAGS='-s -w -static-libgcc -static'
     export EXTRA_LDFLAGS='-s -w -linkmode external -extldflags "-static -lm"'
-    export BUILDTAGS='static netgo osusergo exclude_graphdriver_btrfs seccomp apparmor selinux'
+    export BUILDTAGS='static netgo osusergo exclude_graphdriver_btrfs exclude_graphdriver_devicemapper seccomp apparmor selinux'
     export CGO_ENABLED=1
     export CGO_LDFLAGS='-lgpgme -lassuan -lgpg-error'
     export SOURCE_DATE_EPOCH=0
@@ -42,7 +42,7 @@
     make binaries
   '';
   installPhase = ''
-    install -Dm755 bin/crio $out/bin/crio
-    install -Dm755 bin/pinns $out/bin/pinns
+    install -Dm755 bin/podman $out/bin/podman
+    install -Dm755 bin/podman-remote $out/bin/podman-remote
   '';
 }

```

And then add crun and conmon, in the same way as for [cri-o packaging](https://github.com/cri-o/packaging/blob/main/scripts/bundle/build).

```
/usr/libexec/podman/crun
/usr/libexec/podman/conmon
```
